deployment/deploy.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+set -e
+
+rm -rf build
+
+npm install
+npm run build
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" -o "$TRAVIS_BRANCH" != "$SOURCE_BRANCH" ]; then
+    echo "Skipping deploy; just doing a build."
+    exit 0
+fi
+
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+git config user.name "Travis CI"
+git config user.email "$COMMIT_AUTHOR_EMAIL"
+git checkout --orphan gh-pages
+git rm --cached -r .
+git add assets background.js build icon.png index.html manifest.json worker.js
+git commit -m "Built website from {$SHA}."
+
+ENCRYPTED_KEY_VAR="encrypted_${ENCRYPTION_LABEL}_key"
+ENCRYPTED_IV_VAR="encrypted_${ENCRYPTION_LABEL}_iv"
+ENCRYPTED_KEY=${!ENCRYPTED_KEY_VAR}
+ENCRYPTED_IV=${!ENCRYPTED_IV_VAR}
+openssl aes-256-cbc -K $ENCRYPTED_KEY -iv $ENCRYPTED_IV -in deployment/deploy_key.enc -out deploy_key -d
+chmod 600 deploy_key
+eval `ssh-agent -s`
+ssh-add deploy_key
+
+git push -f "${SSH_REPO}" gh-pages
```

Comment:
Maybe it is worth making this optional and controlled by a config variable so that the build process can be triggered for PRs.